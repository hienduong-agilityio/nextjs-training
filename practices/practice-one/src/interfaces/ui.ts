export interface IInputGroupProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: React.ReactNode;
  customClass?: {
    container?: string;
    inputContainer?: string;
    input?: string;
    button?: string;
  };
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}
