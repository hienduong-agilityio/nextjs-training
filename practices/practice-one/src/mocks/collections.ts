import { Category } from '@/types/category';

type HotDeal = {
  name: Category | 'all';
  count: number;
};

export const HOT_DEALS: HotDeal[] = [
  { name: 'all', count: 2 },
  { name: 'beauty', count: 2 },
  { name: 'fragrances', count: 48 },
  { name: 'furniture', count: 95 },
  { name: 'groceries', count: 23 },
];

export const BRANDS = [
  { name: 'Nike', count: 99 },
  { name: 'Adidas', count: 99 },
  { name: 'Siemens', count: 99 },
];

export const FILTER_GROUP = [{ title: 'Hot Deals', items: HOT_DEALS }];
