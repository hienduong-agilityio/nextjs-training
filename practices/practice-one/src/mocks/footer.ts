import { ROUTE } from '@/constants';

export const FOOTER_NAVIGATION = [
  {
    title: 'Information',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
  {
    title: 'Service',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
  {
    title: 'My Account',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
  {
    title: 'Our Offers',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
];
