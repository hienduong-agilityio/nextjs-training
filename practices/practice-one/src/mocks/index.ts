export * from './product';
export * from './service';
export * from './footer';
export * from './collections';
