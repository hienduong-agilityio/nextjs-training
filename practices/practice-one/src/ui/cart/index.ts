export * from './CartTable';
export * from './OrderSummary';
export * from './VoucherInput';
export * from './EmptyCart';

export { default as CartItemRow } from './CartItemRow';
