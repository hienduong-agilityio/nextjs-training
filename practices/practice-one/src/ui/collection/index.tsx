export * from './LoadingCategoryGroup';
export * from './LoadingFilterSortBar';
