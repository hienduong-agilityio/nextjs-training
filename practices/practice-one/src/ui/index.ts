export * from './products/index';
export * from './collection/index';
