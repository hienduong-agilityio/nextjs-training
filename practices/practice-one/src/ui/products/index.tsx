export * from './CartAndFavoriteActions';
export * from './ProductImages';
export * from './ProductInfo';
export { default as QuantityControl } from './QuantityControl';
export * from './SocialShare';
export * from './RelatedProducts';
export * from './ProductDetailTabs';
export * from './LoadingRelatedProducts';
export * from './LoadingProductCard';
export * from './LoadingProductTabs';
export * from './ProductReview';
