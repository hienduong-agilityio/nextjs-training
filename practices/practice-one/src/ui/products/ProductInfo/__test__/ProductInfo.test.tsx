import { render, screen } from '@testing-library/react';

// Types
import type { IProductProps } from '@/interfaces';

// UI
import { ProductInfo } from '@/ui';

// Mocks
import { PRODUCTS_DATA } from '@/mocks';

describe('ProductInfo Component', () => {
  const product = PRODUCTS_DATA[0];

  const renderProductInfo = (props: IProductProps = product) => {
    return render(<ProductInfo {...props} />);
  };

  it('matches the snapshot', () => {
    const { asFragment } = renderProductInfo();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with product data from PRODUCTS_DATA', () => {
    renderProductInfo();

    expect(screen.getByText(product.name)).toBeInTheDocument();

    if (product.availabilityStatus) {
      expect(screen.getByText('Availability:')).toBeInTheDocument();
      expect(screen.getByText(product.availabilityStatus)).toBeInTheDocument();
    }

    if (product.category) {
      expect(screen.getByText('Category:')).toBeInTheDocument();
      expect(screen.getByText(product.category)).toBeInTheDocument();
    }

    if (product.shippingInformation) {
      expect(screen.getByText(product.shippingInformation)).toBeInTheDocument();
    }

    const reviewLink = screen.getByText('Submit a review');
    expect(reviewLink).toBeInTheDocument();
    expect(reviewLink).toHaveAttribute('href', '/products');
  });
});
