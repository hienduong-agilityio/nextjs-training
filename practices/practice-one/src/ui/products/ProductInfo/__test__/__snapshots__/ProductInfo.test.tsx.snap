// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="mb-4 text-2xl font-semibold text-gray-800 md:text-3xl break-words"
    >
      Sneakers
    </h2>
    <div
      class="flex flex-wrap items-center gap-4 pb-3 border-b-2 border-secondary-1000"
    >
      <div
        class="flex gap-2 space-x-2"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.36363636363636365)"
          />
        </svg>
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.36363636363636365)"
          />
        </svg>
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.36363636363636365)"
          />
        </svg>
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.36363636363636365)"
          />
        </svg>
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.982 17.527 29.044 27.992 32.85 45 20.28 34.725 6.9 44.453l5.169-16.817L0 16.642l15.764-.119L21.688 0l4.574 16.746 15.72.782Z"
            fill="#FFC600"
            transform="scale(0.36363636363636365)"
          />
        </svg>
      </div>
      <span
        class="text-secondary-500 w-max whitespace-nowrap"
      >
        0 reviews
      </span>
      <a
        class="text-sm text-primary-100 hover:underline whitespace-nowrap"
        href="/products"
      >
        Submit a review
      </a>
    </div>
    <div
      class="flex flex-wrap items-center gap-4 py-4"
    >
      <span
        class="text-2xl font-bold text-primary-200"
      >
        $99
      </span>
      <span
        class="text-lg text-gray-500 line-through"
      >
        $129
      </span>
      <span
        class="text-lg font-bold text-danger-50 whitespace-nowrap"
      >
        23% % Off
      </span>
    </div>
    <div
      class="md:w-1/2 space-y-2 text-sm text-gray-600"
    >
      <div
        class="flex justify-between flex-wrap gap-2"
      >
        <span
          class="font-medium whitespace-nowrap"
        >
          Availability:
        </span>
        <span
          class="capitalize"
        >
          unknown
        </span>
      </div>
      <div
        class="flex justify-between flex-wrap gap-2"
      >
        <span
          class="font-medium whitespace-nowrap"
        >
          Category:
        </span>
        <span
          class="capitalize"
        >
          fragrances
        </span>
      </div>
      <p
        class="font-medium"
      >
        Shipping details not available
      </p>
    </div>
  </div>
</DocumentFragment>
`;
